torch/tensor.py

```diff
@@ -77,6 +77,8 @@ def __deepcopy__(self, memo):
                     new_tensor = self.new()
                     new_tensor.set_(new_storage, self.storage_offset(), self.size(), self.stride())
                     new_tensor.requires_grad = self.requires_grad
+            if self.grad is not None:
+                new_tensor.grad = self.grad.clone()
```

Comment:
This should be deepcopied instead of cloned no? Otherwise you might loose things like sharing?

Reply:
fixed